geo/src/algorithm/triangulate_delaunay.rs

```diff
@@ -0,0 +1,645 @@
+                let count = bad_triangle_edge_count.get_mut(idx).unwrap();
+                *count += 1;
+            } else {
+                // These two vectors must be updated at the same time
+                bad_triangle_edges.push(y);
+                bad_triangle_edge_count.push(1);
+            }
+        }
+    }
+
+    // Shared edges are those with a count of > 1
+    let polygon: Vec<Line<T>> = bad_triangle_edge_count
+        .iter()
+        .enumerate()
+        .filter(|(_, count)| **count < 2)
+        .map(|(idx, _)| bad_triangle_edges[idx])
+        .collect();
+
+    // Remove the bad triangles
+    let mut new_triangles: Vec<DelaunayTriangle<T>> = triangles
+        .iter()
+        .filter(|x| !bad_triangles.contains(x))
+        .cloned()
+        .collect();
+
+    polygon
+        .iter()
+        .for_each(|x| new_triangles.push(DelaunayTriangle(Triangle(x.start, x.end, *c))));
+
+    // Replace the triangles
+    triangles.clear();
+    new_triangles.iter().for_each(|x| triangles.push(x.clone()));
+}
+
+fn remove_super_triangle<T: GeoFloat>(
+    triangles: &[DelaunayTriangle<T>],
+    super_triangle: &DelaunayTriangle<T>,
+) -> Vec<DelaunayTriangle<T>> {
+    let mut new_triangles: Vec<DelaunayTriangle<T>> = Vec::new();
+    let super_tri_vertices = super_triangle.0.to_array();
+
+    for tri in triangles.iter() {
+        let mut add_to_update = true;
+        for pt in tri.0.to_array().iter() {
+            if super_tri_vertices.contains(pt) {
+                add_to_update = false;
+            }
+        }
+        if add_to_update {
+            new_triangles.push(tri.clone())
+        }
+    }
+    new_triangles
+}
+
+/// A triangle structure used during Delaunay Triangulation.
+#[derive(Debug, Clone, PartialEq)]
+pub struct DelaunayTriangle<T: GeoFloat>(Triangle<T>);
+
+impl<T: GeoFloat> From<Triangle<T>> for DelaunayTriangle<T> {
+    fn from(value: Triangle<T>) -> Self {
+        DelaunayTriangle(value)
+    }
+}
+
+impl<T: GeoFloat> From<DelaunayTriangle<T>> for Triangle<T> {
+    fn from(value: DelaunayTriangle<T>) -> Self {
+        value.0
+    }
+}
+
+fn is_line_shared<T: GeoFloat>(a: &Line<T>, b: &Line<T>) -> bool {
+    (a.start == b.start && a.end == b.end) || (a.start == b.end && a.end == b.start)
+}
+
+/// Methods required for delaunay triangulation
+impl<T: GeoFloat> DelaunayTriangle<T>
+where
+    f64: From<T>,
+{
+    #[cfg(feature = "voronoi")]
+    /// Check if a `DelaunayTriangle` is a neighbour i.e.
+    /// shares an edge.
+    /// If the triangles are neighbours the shared edge is returned.
+    pub(crate) fn shares_edge(&self, other: &DelaunayTriangle<T>) -> Option<Line<T>> {
+        let other_lines = other.0.to_lines();
+        for line in self.0.to_lines().iter() {
+            for other_line in other_lines.iter() {
+                if is_line_shared(line, other_line) {
+                    return Some(*line);
+                }
+            }
+        }
+        None
```

Comment:
This can be simplified to

```rust
let other_lines = other.0.to_lines();
self.0.to_lines()
  .iter()
  .find(|line| {
    other_lines.iter()
      .any(|other| is_line_shared(line, other))
})
```

Reply:
yes with a couple of modifications to account for the temporary value that is freed while in use